pkg/client/group_mgmt_client.go

```diff
@@ -60,7 +64,53 @@ type Argument struct {
 	JobId string `json:"job_id,omitempty"`
 }
 
-func newGroupMgmtClient(ipAddress, username, password, apiVersion string, waitOnJobs bool) *GroupMgmtClient {
+type ClientOption func(*GroupMgmtClient)
+
+func WithHost(host string) func(*GroupMgmtClient) {
+	return func(groupMgmtClient *GroupMgmtClient) {
+		groupMgmtClient.Host = host
+	}
+}
+
+func WithUser(username string) func(*GroupMgmtClient) {
+	return func(groupMgmtClient *GroupMgmtClient) {
+		groupMgmtClient.Username = username
+		groupMgmtClient.isTenant = false
+	}
+}
+
+func WithTenantUser(username string) func(*GroupMgmtClient) {
+	return func(groupMgmtClient *GroupMgmtClient) {
+		groupMgmtClient.Username = username
+		groupMgmtClient.isTenant = true
+	}
+}
+
+func WithPassword(password string) func(*GroupMgmtClient) {
+	return func(groupMgmtClient *GroupMgmtClient) {
+		groupMgmtClient.Password = password
+	}
+}
+
+func WithApiVersion(version string) func(*GroupMgmtClient) {
```

Comment:
Somewhere we need to define the default value as "v1" (which makes use of this function pretty unnecessary).

Reply:
hmm I'm thinking of defining the default value inside of `newGroupMgmtClient()` (see line 121-128). But if we have more than 1 version, wouldn't this function still be necessary? (for example, we set the default to be v1 but users can opt to use v2 instead)?